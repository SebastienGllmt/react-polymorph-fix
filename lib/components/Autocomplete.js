'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Autocomplete = undefined;

var _startsWith2 = require('lodash/startsWith');

var _startsWith3 = _interopRequireDefault(_startsWith2);

var _some2 = require('lodash/some');

var _some3 = _interopRequireDefault(_some2);

var _pullAt2 = require('lodash/pullAt');

var _pullAt3 = _interopRequireDefault(_pullAt2);

var _concat2 = require('lodash/concat');

var _concat3 = _interopRequireDefault(_concat2);

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _createRef = require('create-react-ref/lib/createRef');

var _createRef2 = _interopRequireDefault(_createRef);

var _GlobalListeners = require('./HOC/GlobalListeners');

var _withTheme = require('./HOC/withTheme');

var _themes = require('../utils/themes');

var _props3 = require('../utils/props');

var _2 = require('.');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

// external libraries


// interal components


// internal utility functions


var AutocompleteBase = function (_Component) {
  _inherits(AutocompleteBase, _Component);

  // define static properties

  // declare ref types
  function AutocompleteBase(props) {
    _classCallCheck(this, AutocompleteBase);

    // define refs
    var _this = _possibleConstructorReturn(this, (AutocompleteBase.__proto__ || Object.getPrototypeOf(AutocompleteBase)).call(this, props));

    _initialiseProps.call(_this);

    _this.rootElement = (0, _createRef2.default)();
    _this.inputElement = (0, _createRef2.default)();
    _this.suggestionsElement = (0, _createRef2.default)();
    _this.optionsElement = (0, _createRef2.default)();

    var context = props.context,
        themeId = props.themeId,
        theme = props.theme,
        themeOverrides = props.themeOverrides,
        sortAlphabetically = props.sortAlphabetically,
        options = props.options,
        preselectedOptions = props.preselectedOptions;


    _this.state = {
      inputValue: '',
      error: '',
      selectedOptions: preselectedOptions || [],
      filteredOptions: sortAlphabetically && options ? options.sort() : options || [],
      isOpen: false,
      mouseIsOverOptions: false,
      composedTheme: (0, _themes.composeTheme)((0, _themes.addThemeId)(theme || context.theme, themeId), (0, _themes.addThemeId)(themeOverrides, themeId), context.ROOT_THEME_API)
    };
    return _this;
  }

  _createClass(AutocompleteBase, [{
    key: 'componentWillReceiveProps',
    value: function componentWillReceiveProps(nextProps) {
      (0, _themes.didThemePropsChange)(this.props, nextProps, this.setState.bind(this));
    }

    // onChange handler for input element in AutocompleteSkin


    // passed to Options onChange handler in AutocompleteSkin


    // returns an object containing props, theme, and method handlers
    // associated with rendering this.state.selectedOptions, the user can call
    // this in the body of the renderSelections function

  }, {
    key: 'render',
    value: function render() {
      var _this2 = this;

      // destructuring props ensures only the "...rest" get passed down
      var _props = this.props,
          context = _props.context,
          invalidCharsRegex = _props.invalidCharsRegex,
          multipleSameSelections = _props.multipleSameSelections,
          sortAlphabetically = _props.sortAlphabetically,
          skin = _props.skin,
          theme = _props.theme,
          themeOverrides = _props.themeOverrides,
          onChange = _props.onChange,
          error = _props.error,
          rest = _objectWithoutProperties(_props, ['context', 'invalidCharsRegex', 'multipleSameSelections', 'sortAlphabetically', 'skin', 'theme', 'themeOverrides', 'onChange', 'error']);

      var AutocompleteSkin = skin || context.skins[_2.IDENTIFIERS.AUTOCOMPLETE];

      return _react2.default.createElement(
        _GlobalListeners.GlobalListeners,
        {
          mouseIsOverOptions: this.state.mouseIsOverOptions,
          optionsIsOpen: this.state.isOpen,
          optionsIsOpeningUpward: this.props.isOpeningUpward,
          optionsRef: this.optionsElement,
          rootRef: this.rootElement,
          toggleOpen: this.toggleOpen
        },
        function (_ref) {
          var optionsMaxHeight = _ref.optionsMaxHeight;
          return _react2.default.createElement(AutocompleteSkin, _extends({
            error: error || _this2.state.error,
            filteredOptions: _this2.state.filteredOptions,
            getSelectionProps: _this2.getSelectionProps,
            handleAutocompleteClick: _this2.handleAutocompleteClick,
            handleChange: _this2.handleChange,
            handleInputChange: _this2.handleInputChange,
            inputRef: _this2.inputElement,
            inputValue: _this2.state.inputValue,
            isOpen: _this2.state.isOpen,
            onKeyDown: _this2.onKeyDown,
            optionsMaxHeight: optionsMaxHeight,
            optionsRef: _this2.optionsElement,
            removeOption: _this2.removeOption,
            rootRef: _this2.rootElement,
            selectedOptions: _this2.state.selectedOptions,
            suggestionsRef: _this2.suggestionsElement,
            theme: _this2.state.composedTheme,
            toggleMouseLocation: _this2.toggleMouseLocation,
            toggleOpen: _this2.toggleOpen
          }, rest));
        }
      );
    }

    // ======== PRIVATE METHOD ==========

  }]);

  return AutocompleteBase;
}(_react.Component);

AutocompleteBase.displayName = 'Autocomplete';
AutocompleteBase.defaultProps = {
  context: (0, _withTheme.createEmptyContext)(),
  error: null,
  invalidCharsRegex: /[^a-zA-Z0-9]/g, // only allow letters and numbers by default
  isOpeningUpward: false,
  maxVisibleOptions: 10, // max number of visible options
  multipleSameSelections: true, // if true then same word can be selected multiple times
  options: [],
  sortAlphabetically: true, // options are sorted alphabetically by default
  theme: null,
  themeId: _2.IDENTIFIERS.AUTOCOMPLETE,
  themeOverrides: {}
};

var _initialiseProps = function _initialiseProps() {
  var _this3 = this;

  this.clear = function () {
    return _this3._removeOptions();
  };

  this.focus = function () {
    return _this3.handleAutocompleteClick();
  };

  this.open = function () {
    return _this3.setState({ isOpen: true });
  };

  this.close = function () {
    return _this3.setState({ isOpen: false });
  };

  this.toggleOpen = function () {
    if (_this3.state.isOpen && _this3.optionsElement && _this3.optionsElement.current) {
      // set Options scroll position to top on close
      _this3.optionsElement.current.scrollTop = 0;
    }
    _this3.setState({ isOpen: !_this3.state.isOpen });
  };

  this.toggleMouseLocation = function () {
    return _this3.setState({ mouseIsOverOptions: !_this3.state.mouseIsOverOptions });
  };

  this.handleAutocompleteClick = function () {
    var inputElement = _this3.inputElement;

    if (inputElement && inputElement.current) {
      inputElement.current.focus();
    }
    // toggle options open/closed
    _this3.toggleOpen();
  };

  this.onKeyDown = function (event) {

    if ( // Check for backspace in order to delete the last selected option
    event.keyCode === 8 && !event.target.value && _this3.state.selectedOptions.length) {
      // Remove last selected option
      _this3.removeOption(_this3.state.selectedOptions.length - 1, event);
    } else if (event.keyCode === 27) {
      // ESCAPE key: Stops propagation & modal closing
      event.stopPropagation();
    } else if (event.keyCode === 13) {
      // ENTER key: Opens suggestions
      _this3.open();
    }
  };

  this.handleInputChange = function (event) {
    _this3._setInputValue(event.target.value);
  };

  this.handleChange = function (option, event) {
    _this3.updateSelectedOptions(event, option);
  };

  this.updateSelectedOptions = function (event) {
    var selectedOption = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : null;
    var _props2 = _this3.props,
        maxSelections = _props2.maxSelections,
        multipleSameSelections = _props2.multipleSameSelections;
    var _state = _this3.state,
        selectedOptions = _state.selectedOptions,
        filteredOptions = _state.filteredOptions,
        isOpen = _state.isOpen;

    var canMoreOptionsBeSelected = maxSelections != null ? selectedOptions.length < maxSelections : true;
    var areFilteredOptionsAvailable = filteredOptions && filteredOptions.length > 0;

    if (!maxSelections || canMoreOptionsBeSelected && areFilteredOptionsAvailable) {
      if (!selectedOption) return;
      var option = selectedOption.trim();
      var optionCanBeSelected = selectedOptions.indexOf(option) < 0 && !multipleSameSelections || multipleSameSelections;

      if (option && optionCanBeSelected && isOpen) {
        var newSelectedOptions = (0, _concat3.default)(selectedOptions, option);
        _this3.selectionChanged(newSelectedOptions, event);
        _this3.setState({ selectedOptions: newSelectedOptions, isOpen: false });
      }
    }

    _this3._setInputValue('');
  };

  this.removeOption = function (index, event) {
    var selectedOptions = _this3.state.selectedOptions;
    (0, _pullAt3.default)(selectedOptions, index);
    _this3.selectionChanged(selectedOptions, event);
    _this3.setState({ selectedOptions: selectedOptions });
  };

  this.selectionChanged = function (selectedOptions, event) {
    if (_this3.props.onChange) _this3.props.onChange(selectedOptions, event);
  };

  this.getSelectionProps = function () {
    var _ref2 = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : {},
        _removeSelection = _ref2.removeSelection;

    var themeId = _this3.props.themeId;
    var _state2 = _this3.state,
        inputValue = _state2.inputValue,
        isOpen = _state2.isOpen,
        selectedOptions = _state2.selectedOptions,
        composedTheme = _state2.composedTheme;

    return {
      inputValue: inputValue,
      isOpen: isOpen,
      selectedOptions: selectedOptions,
      theme: composedTheme[themeId],
      removeSelection: function removeSelection(index, event
      // the user's custom removeSelection event handler is composed with
      // the internal functionality of Autocomplete (this.removeOption)
      ) {
        return (0, _props3.composeFunctions)(_removeSelection, _this3.removeOption)(index, event);
      }
    };
  };

  this._removeOptions = function () {
    var onChange = _this3.props.onChange;

    onChange ? onChange([]) : null;
    _this3.setState({ selectedOptions: [], inputValue: '' });
  };

  this._filterOptions = function (value) {
    var filteredOptions = [];

    if (value !== '') {
      (0, _some3.default)(_this3.props.options, function (option) {
        if ((0, _startsWith3.default)(option, value)) {
          filteredOptions.push(option);
        }
      });
    } else {
      filteredOptions = _this3.props.options;
    }

    return filteredOptions;
  };

  this._filterInvalidChars = function (value) {
    var filteredValue = '';

    if (_this3.props.invalidCharsRegex.test(value)) {
      filteredValue = value.replace(_this3.props.invalidCharsRegex, '');
    } else {
      filteredValue = value;
    }

    return filteredValue;
  };

  this._setInputValue = function (value) {
    var filteredValue = _this3._filterInvalidChars(value);
    var filteredOptions = _this3._filterOptions(filteredValue);
    _this3.setState({
      isOpen: true,
      inputValue: filteredValue,
      filteredOptions: filteredOptions
    });
  };
};

var Autocomplete = exports.Autocomplete = (0, _withTheme.withTheme)(AutocompleteBase);
//# sourceMappingURL=Autocomplete.js.map