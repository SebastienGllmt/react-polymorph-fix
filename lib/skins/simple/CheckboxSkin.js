'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.CheckboxSkin = undefined;

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };
// external libraries


// internal utility functions


var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

var _props = require('../../utils/props');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var CheckboxSkin = exports.CheckboxSkin = function CheckboxSkin(props) {
  return _react2.default.createElement(
    'div',
    {
      role: 'presentation',
      'aria-hidden': true,
      className: (0, _classnames2.default)([props.className, props.theme[props.themeId].root, props.disabled ? props.theme[props.themeId].disabled : null, props.checked ? props.theme[props.themeId].checked : null]),
      onClick: function onClick(event) {
        if (!props.disabled && props.onChange) {
          props.onChange(!props.checked, event);
        }
      }
    },
    _react2.default.createElement('input', _extends({}, (0, _props.pickDOMProps)(props), {
      className: props.theme[props.themeId].input,
      type: 'checkbox'
    })),
    _react2.default.createElement('div', {
      className: (0, _classnames2.default)([props.theme[props.themeId].check, props.checked ? props.theme[props.themeId].checked : null])
    }),
    props.label && _react2.default.createElement(
      'label',
      { className: props.theme[props.themeId].label },
      props.label
    )
  );
};
//# sourceMappingURL=CheckboxSkin.js.map