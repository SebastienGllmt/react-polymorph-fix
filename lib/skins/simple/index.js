'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.SimpleSkins = undefined;

var _SimpleSkins;

var _components = require('../../components');

var _AutocompleteSkin = require('./AutocompleteSkin');

var _BubbleSkin = require('./BubbleSkin');

var _ButtonSkin = require('./ButtonSkin');

var _CheckboxSkin = require('./CheckboxSkin');

var _FormFieldSkin = require('./FormFieldSkin');

var _HeaderSkin = require('./HeaderSkin');

var _InfiniteScrollSkin = require('./InfiniteScrollSkin');

var _InputSkin = require('./InputSkin');

var _LoadingSpinnerSkin = require('./LoadingSpinnerSkin');

var _ModalSkin = require('./ModalSkin');

var _OptionsSkin = require('./OptionsSkin');

var _ProgressBarSkin = require('./ProgressBarSkin');

var _RadioSkin = require('./RadioSkin');

var _SelectSkin = require('./SelectSkin');

var _SwitchSkin = require('./SwitchSkin');

var _TextAreaSkin = require('./TextAreaSkin');

var _TogglerSkin = require('./TogglerSkin');

var _TooltipSkin = require('./TooltipSkin');

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

var SimpleSkins = exports.SimpleSkins = (_SimpleSkins = {}, _defineProperty(_SimpleSkins, _components.IDENTIFIERS.AUTOCOMPLETE, _AutocompleteSkin.AutocompleteSkin), _defineProperty(_SimpleSkins, _components.IDENTIFIERS.BUBBLE, _BubbleSkin.BubbleSkin), _defineProperty(_SimpleSkins, _components.IDENTIFIERS.BUTTON, _ButtonSkin.ButtonSkin), _defineProperty(_SimpleSkins, _components.IDENTIFIERS.CHECKBOX, _CheckboxSkin.CheckboxSkin), _defineProperty(_SimpleSkins, _components.IDENTIFIERS.FORM_FIELD, _FormFieldSkin.FormFieldSkin), _defineProperty(_SimpleSkins, _components.IDENTIFIERS.HEADER, _HeaderSkin.HeaderSkin), _defineProperty(_SimpleSkins, _components.IDENTIFIERS.INFINITE_SCROLL, _InfiniteScrollSkin.InfiniteScrollSkin), _defineProperty(_SimpleSkins, _components.IDENTIFIERS.INPUT, _InputSkin.InputSkin), _defineProperty(_SimpleSkins, _components.IDENTIFIERS.LOADING_SPINNER, _LoadingSpinnerSkin.LoadingSpinnerSkin), _defineProperty(_SimpleSkins, _components.IDENTIFIERS.MODAL, _ModalSkin.ModalSkin), _defineProperty(_SimpleSkins, _components.IDENTIFIERS.OPTIONS, _OptionsSkin.OptionsSkin), _defineProperty(_SimpleSkins, _components.IDENTIFIERS.PROGRESS_BAR, _ProgressBarSkin.ProgressBarSkin), _defineProperty(_SimpleSkins, _components.IDENTIFIERS.RADIO, _RadioSkin.RadioSkin), _defineProperty(_SimpleSkins, _components.IDENTIFIERS.SELECT, _SelectSkin.SelectSkin), _defineProperty(_SimpleSkins, _components.IDENTIFIERS.SWITCH, _SwitchSkin.SwitchSkin), _defineProperty(_SimpleSkins, _components.IDENTIFIERS.TEXT_AREA, _TextAreaSkin.TextAreaSkin), _defineProperty(_SimpleSkins, _components.IDENTIFIERS.TOGGLER, _TogglerSkin.TogglerSkin), _defineProperty(_SimpleSkins, _components.IDENTIFIERS.TOOLTIP, _TooltipSkin.TooltipSkin), _SimpleSkins);
//# sourceMappingURL=index.js.map