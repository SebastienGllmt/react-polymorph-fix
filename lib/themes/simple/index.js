'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.SimpleTheme = undefined;

var _SimpleTheme;

var _components = require('../../components');

var _SimpleAutocomplete = require('./SimpleAutocomplete.scss');

var _SimpleAutocomplete2 = _interopRequireDefault(_SimpleAutocomplete);

var _SimpleBubble = require('./SimpleBubble.scss');

var _SimpleBubble2 = _interopRequireDefault(_SimpleBubble);

var _SimpleButton = require('./SimpleButton.scss');

var _SimpleButton2 = _interopRequireDefault(_SimpleButton);

var _SimpleCheckbox = require('./SimpleCheckbox.scss');

var _SimpleCheckbox2 = _interopRequireDefault(_SimpleCheckbox);

var _SimpleFlex = require('./SimpleFlex.scss');

var _SimpleFlex2 = _interopRequireDefault(_SimpleFlex);

var _SimpleFormField = require('./SimpleFormField.scss');

var _SimpleFormField2 = _interopRequireDefault(_SimpleFormField);

var _SimpleGrid = require('./SimpleGrid.scss');

var _SimpleGrid2 = _interopRequireDefault(_SimpleGrid);

var _SimpleGutter = require('./SimpleGutter.scss');

var _SimpleGutter2 = _interopRequireDefault(_SimpleGutter);

var _SimpleHeader = require('./SimpleHeader.scss');

var _SimpleHeader2 = _interopRequireDefault(_SimpleHeader);

var _SimpleInfiniteScroll = require('./SimpleInfiniteScroll.scss');

var _SimpleInfiniteScroll2 = _interopRequireDefault(_SimpleInfiniteScroll);

var _SimpleInput = require('./SimpleInput.scss');

var _SimpleInput2 = _interopRequireDefault(_SimpleInput);

var _SimpleLoadingSpinner = require('./SimpleLoadingSpinner.scss');

var _SimpleLoadingSpinner2 = _interopRequireDefault(_SimpleLoadingSpinner);

var _SimpleModal = require('./SimpleModal.scss');

var _SimpleModal2 = _interopRequireDefault(_SimpleModal);

var _SimpleOptions = require('./SimpleOptions.scss');

var _SimpleOptions2 = _interopRequireDefault(_SimpleOptions);

var _SimpleProgressBar = require('./SimpleProgressBar.scss');

var _SimpleProgressBar2 = _interopRequireDefault(_SimpleProgressBar);

var _SimpleRadio = require('./SimpleRadio.scss');

var _SimpleRadio2 = _interopRequireDefault(_SimpleRadio);

var _SimpleSelect = require('./SimpleSelect.scss');

var _SimpleSelect2 = _interopRequireDefault(_SimpleSelect);

var _SimpleSwitch = require('./SimpleSwitch.scss');

var _SimpleSwitch2 = _interopRequireDefault(_SimpleSwitch);

var _SimpleTextArea = require('./SimpleTextArea.scss');

var _SimpleTextArea2 = _interopRequireDefault(_SimpleTextArea);

var _SimpleToggler = require('./SimpleToggler.scss');

var _SimpleToggler2 = _interopRequireDefault(_SimpleToggler);

var _SimpleTooltip = require('./SimpleTooltip.scss');

var _SimpleTooltip2 = _interopRequireDefault(_SimpleTooltip);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }
// import theme IDENTIFIERS constants


// css modules plugin converts all imports below into plain objects


// SimpleTheme is a plain object serving as the default export.
// Each key is named after a component and each key's value
// is the component's corresponding theme. The user can
// pass this entire obj directly to ThemeProvider via the "theme" prop
var SimpleTheme = exports.SimpleTheme = (_SimpleTheme = {}, _defineProperty(_SimpleTheme, _components.IDENTIFIERS.AUTOCOMPLETE, _SimpleAutocomplete2.default), _defineProperty(_SimpleTheme, _components.IDENTIFIERS.BUBBLE, _SimpleBubble2.default), _defineProperty(_SimpleTheme, _components.IDENTIFIERS.BUTTON, _SimpleButton2.default), _defineProperty(_SimpleTheme, _components.IDENTIFIERS.CHECKBOX, _SimpleCheckbox2.default), _defineProperty(_SimpleTheme, _components.IDENTIFIERS.FLEX, _SimpleFlex2.default), _defineProperty(_SimpleTheme, _components.IDENTIFIERS.FORM_FIELD, _SimpleFormField2.default), _defineProperty(_SimpleTheme, _components.IDENTIFIERS.GRID, _SimpleGrid2.default), _defineProperty(_SimpleTheme, _components.IDENTIFIERS.GUTTER, _SimpleGutter2.default), _defineProperty(_SimpleTheme, _components.IDENTIFIERS.HEADER, _SimpleHeader2.default), _defineProperty(_SimpleTheme, _components.IDENTIFIERS.INFINITE_SCROLL, _SimpleInfiniteScroll2.default), _defineProperty(_SimpleTheme, _components.IDENTIFIERS.INPUT, _SimpleInput2.default), _defineProperty(_SimpleTheme, _components.IDENTIFIERS.LOADING_SPINNER, _SimpleLoadingSpinner2.default), _defineProperty(_SimpleTheme, _components.IDENTIFIERS.MODAL, _SimpleModal2.default), _defineProperty(_SimpleTheme, _components.IDENTIFIERS.OPTIONS, _SimpleOptions2.default), _defineProperty(_SimpleTheme, _components.IDENTIFIERS.PROGRESS_BAR, _SimpleProgressBar2.default), _defineProperty(_SimpleTheme, _components.IDENTIFIERS.RADIO, _SimpleRadio2.default), _defineProperty(_SimpleTheme, _components.IDENTIFIERS.SELECT, _SimpleSelect2.default), _defineProperty(_SimpleTheme, _components.IDENTIFIERS.SWITCH, _SimpleSwitch2.default), _defineProperty(_SimpleTheme, _components.IDENTIFIERS.TEXT_AREA, _SimpleTextArea2.default), _defineProperty(_SimpleTheme, _components.IDENTIFIERS.TOGGLER, _SimpleToggler2.default), _defineProperty(_SimpleTheme, _components.IDENTIFIERS.TOOLTIP, _SimpleTooltip2.default), _SimpleTheme);
//# sourceMappingURL=index.js.map