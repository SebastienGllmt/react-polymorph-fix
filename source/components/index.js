// @flow
export const IDENTIFIERS = {
  AUTOCOMPLETE: 'autocomplete',
  BUBBLE: 'bubble',
  BUTTON: 'button',
  CHECKBOX: 'checkbox',
  FLEX: 'flex',
  FORM_FIELD: 'formfield',
  GRID: 'grid',
  GUTTER: 'gutter',
  HEADER: 'header',
  INFINITE_SCROLL: 'infinitescroll',
  INPUT: 'input',
  LOADING_SPINNER: 'loadingspinner',
  MODAL: 'modal',
  OPTIONS: 'options',
  PROGRESS_BAR: 'progressbar',
  RADIO: 'radio',
  SELECT: 'select',
  SWITCH: 'switch',
  TEXT_AREA: 'textarea',
  TOGGLER: 'toggler',
  TOOLTIP: 'tooltip'
};
