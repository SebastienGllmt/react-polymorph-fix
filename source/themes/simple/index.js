// @flow
// import theme IDENTIFIERS constants
import { IDENTIFIERS } from '../../components';

// css modules plugin converts all imports below into plain objects
import SimpleAutocomplete from './SimpleAutocomplete.scss';
import SimpleBubble from './SimpleBubble.scss';
import SimpleButton from './SimpleButton.scss';
import SimpleCheckbox from './SimpleCheckbox.scss';
import SimpleFlex from './SimpleFlex.scss';
import SimpleFormField from './SimpleFormField.scss';
import SimpleGrid from './SimpleGrid.scss';
import SimpleGutter from './SimpleGutter.scss';
import SimpleHeader from './SimpleHeader.scss';
import SimpleInfiniteScroll from './SimpleInfiniteScroll.scss';
import SimpleInput from './SimpleInput.scss';
import SimpleLoadingSpinner from './SimpleLoadingSpinner.scss';
import SimpleModal from './SimpleModal.scss';
import SimpleOptions from './SimpleOptions.scss';
import SimpleProgressBar from './SimpleProgressBar.scss';
import SimpleRadio from './SimpleRadio.scss';
import SimpleSelect from './SimpleSelect.scss';
import SimpleSwitch from './SimpleSwitch.scss';
import SimpleTextArea from './SimpleTextArea.scss';
import SimpleToggler from './SimpleToggler.scss';
import SimpleTooltip from './SimpleTooltip.scss';

// SimpleTheme is a plain object serving as the default export.
// Each key is named after a component and each key's value
// is the component's corresponding theme. The user can
// pass this entire obj directly to ThemeProvider via the "theme" prop
export const SimpleTheme = {
  [IDENTIFIERS.AUTOCOMPLETE]: SimpleAutocomplete,
  [IDENTIFIERS.BUBBLE]: SimpleBubble,
  [IDENTIFIERS.BUTTON]: SimpleButton,
  [IDENTIFIERS.CHECKBOX]: SimpleCheckbox,
  [IDENTIFIERS.FLEX]: SimpleFlex,
  [IDENTIFIERS.FORM_FIELD]: SimpleFormField,
  [IDENTIFIERS.GRID]: SimpleGrid,
  [IDENTIFIERS.GUTTER]: SimpleGutter,
  [IDENTIFIERS.HEADER]: SimpleHeader,
  [IDENTIFIERS.INFINITE_SCROLL]: SimpleInfiniteScroll,
  [IDENTIFIERS.INPUT]: SimpleInput,
  [IDENTIFIERS.LOADING_SPINNER]: SimpleLoadingSpinner,
  [IDENTIFIERS.MODAL]: SimpleModal,
  [IDENTIFIERS.OPTIONS]: SimpleOptions,
  [IDENTIFIERS.PROGRESS_BAR]: SimpleProgressBar,
  [IDENTIFIERS.RADIO]: SimpleRadio,
  [IDENTIFIERS.SELECT]: SimpleSelect,
  [IDENTIFIERS.SWITCH]: SimpleSwitch,
  [IDENTIFIERS.TEXT_AREA]: SimpleTextArea,
  [IDENTIFIERS.TOGGLER]: SimpleToggler,
  [IDENTIFIERS.TOOLTIP]: SimpleTooltip
};
