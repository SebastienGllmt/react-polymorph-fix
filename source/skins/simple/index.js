// @flow
import { IDENTIFIERS } from '../../components';
import { AutocompleteSkin } from './AutocompleteSkin';
import { BubbleSkin } from './BubbleSkin';
import { ButtonSkin } from './ButtonSkin';
import { CheckboxSkin } from './CheckboxSkin';
import { FormFieldSkin } from './FormFieldSkin';
import { HeaderSkin } from './HeaderSkin';
import { InfiniteScrollSkin } from './InfiniteScrollSkin';
import { InputSkin } from './InputSkin';
import { LoadingSpinnerSkin } from './LoadingSpinnerSkin';
import { ModalSkin } from './ModalSkin';
import { OptionsSkin } from './OptionsSkin';
import { ProgressBarSkin } from './ProgressBarSkin';
import { RadioSkin } from './RadioSkin';
import { SelectSkin } from './SelectSkin';
import { SwitchSkin } from './SwitchSkin';
import { TextAreaSkin } from './TextAreaSkin';
import { TogglerSkin } from './TogglerSkin';
import { TooltipSkin } from './TooltipSkin';

export const SimpleSkins = {
  [IDENTIFIERS.AUTOCOMPLETE]: AutocompleteSkin,
  [IDENTIFIERS.BUBBLE]: BubbleSkin,
  [IDENTIFIERS.BUTTON]: ButtonSkin,
  [IDENTIFIERS.CHECKBOX]: CheckboxSkin,
  [IDENTIFIERS.FORM_FIELD]: FormFieldSkin,
  [IDENTIFIERS.HEADER]: HeaderSkin,
  [IDENTIFIERS.INFINITE_SCROLL]: InfiniteScrollSkin,
  [IDENTIFIERS.INPUT]: InputSkin,
  [IDENTIFIERS.LOADING_SPINNER]: LoadingSpinnerSkin,
  [IDENTIFIERS.MODAL]: ModalSkin,
  [IDENTIFIERS.OPTIONS]: OptionsSkin,
  [IDENTIFIERS.PROGRESS_BAR]: ProgressBarSkin,
  [IDENTIFIERS.RADIO]: RadioSkin,
  [IDENTIFIERS.SELECT]: SelectSkin,
  [IDENTIFIERS.SWITCH]: SwitchSkin,
  [IDENTIFIERS.TEXT_AREA]: TextAreaSkin,
  [IDENTIFIERS.TOGGLER]: TogglerSkin,
  [IDENTIFIERS.TOOLTIP]: TooltipSkin
};
