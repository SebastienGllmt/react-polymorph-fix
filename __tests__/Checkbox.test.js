import React from 'react';

import { Checkbox } from '../source/components/Checkbox';
import { CheckboxSkin } from '../source/skins/simple/CheckboxSkin';
import { renderInSimpleTheme } from './helpers/theming';

test('Checkbox renders correctly', () => {
  const component = renderInSimpleTheme(
    <Checkbox skin={CheckboxSkin} />
  );

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Checkbox renders with a label', () => {
  const component = renderInSimpleTheme(
    <Checkbox label="check here" skin={CheckboxSkin} />
  );

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Checkbox is disabled', () => {
  const component = renderInSimpleTheme(
    <Checkbox disabled skin={CheckboxSkin} />
  );

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Checkbox is checked', () => {
  const component = renderInSimpleTheme(
    <Checkbox checked skin={CheckboxSkin} />
  );

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
