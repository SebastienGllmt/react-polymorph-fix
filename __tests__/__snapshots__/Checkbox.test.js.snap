// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox is checked 1`] = `
<div
  aria-hidden={true}
  className="root checked"
  onClick={[Function]}
  role="presentation"
>
  <input
    checked={true}
    className="input"
    type="checkbox"
  />
  <div
    className="check checked"
  />
</div>
`;

exports[`Checkbox is disabled 1`] = `
<div
  aria-hidden={true}
  className="root disabled"
  onClick={[Function]}
  role="presentation"
>
  <input
    checked={false}
    className="input"
    disabled={true}
    type="checkbox"
  />
  <div
    className="check"
  />
</div>
`;

exports[`Checkbox renders correctly 1`] = `
<div
  aria-hidden={true}
  className="root"
  onClick={[Function]}
  role="presentation"
>
  <input
    checked={false}
    className="input"
    type="checkbox"
  />
  <div
    className="check"
  />
</div>
`;

exports[`Checkbox renders with a label 1`] = `
<div
  aria-hidden={true}
  className="root"
  onClick={[Function]}
  role="presentation"
>
  <input
    checked={false}
    className="input"
    label="check here"
    type="checkbox"
  />
  <div
    className="check"
  />
  <label
    className="label"
  >
    check here
  </label>
</div>
`;
